NetworthManager.js

```diff
@@ -0,0 +1,207 @@
+     * @param {number} [options.itemsRetries=3] - The amount of retries to fetch the items when failing to fetch them
+     * @param {number} [options.itemsInterval=1000 * 60 * 60 * 12] - The interval to fetch the items from the Hypixel API
+     * @param {boolean} [options.onlyNetworth=false] - Whether to only return the total networth or the items as well
+     * @param {boolean} [options.stackItems=true] - Whether to stack items with the same name and price
+     * @param {boolean} [options.includeItemData=false] - Whether to include the item data as a property in the item object
+     */
+    constructor({ networthType, cachePrices, pricesRetries, itemsRetries, itemsInterval, onlyNetworth, stackItems, includeItemData }) {
+        this.networthType = networthType || NetworthTypes.Normal;
+        this.cachePrices = cachePrices || true;
+        this.pricesRetries = pricesRetries || 3;
+        this.itemsRetries = itemsRetries || 3;
+        this.itemsInterval = itemsInterval || 1000 * 60 * 60 * 12; // 12 hours
+        this.onlyNetworth = onlyNetworth || false;
+        this.stackItems = stackItems || true;
+        this.includeItemData = includeItemData || false;
+
+        this.itemsPromise = updateItems(this.itemsRetries);
+        setInterval(() => {
+            updateItems(this.itemsRetries);
+        }, this.itemsInterval);
+    }
+
+    /**
+     * Set the type of networth calculation to use
+     * @param {NetworthTypes} networthType - The type of networth calculation to use
+     * @returns {NetworthManager} The NetworthManager instance
+     */
+    setNetworthType(networthType) {
+        this.networthType = networthType;
+        return this;
+    }
+
+    /**
+     * Returns the networth of a profile
+     * @param {object} profileData - The profile player data from the Hypixel API (profile.members[uuid])
+     * @param {object} [museumData] - The museum data from the Hypixel API (museum.members[uuid]). If not provided, the museum data will not be included in the networth calculation
+     * @param {number} bankBalance - The bank balance of the player from the Hypixel API (profile.banking.balance)
+     * @param {object} [prices] - A prices object generated from the getPrices function. If not provided, the prices will be retrieved every time the function is called
+     * @returns An object containing the player's networth calculation
+     */
+    async getNetworth({ profileData, museumData, bankBalance, prices: prePrices }) {
+        if (!profileData) throw new NetworthError('Invalid profile data provided');
+        const purse = profileData.currencies?.coin_purse;
+        const personalBankBalance = profileData.profile?.bank_account;
+        const prices = await parsePrices(prePrices, this.cachePrices, this.pricesRetries);
+        const items = await parseItems(profileData, museumData);
+        await this.itemsPromise.catch(() => {});
```

Comment:
??

Reply:
When the manager is created, it instantly tries to get the items from the skyblock items endpoint in the Hypixel API. And in line 31 you can see:
`this.itemPromise = updateItems(this.itemRetries)`
what line 61 basically does is wait for the items to be there from the Hypixel API but still continue if it fails thats why the catch is added.
This is just when you call it for the first time because it needs to get the items from the Hypixel API first

Reply:
Ic, ic